extensions/sql-bindings/src/test/testContext.ts

```diff
@@ -0,0 +1,382 @@
+		ignoreExtendedProperties: false,
+		ignoreDmlTriggerState: false,
+		ignoreDmlTriggerOrder: false,
+		ignoreDefaultSchema: false,
+		ignoreDdlTriggerState: false,
+		ignoreDdlTriggerOrder: false,
+		ignoreCryptographicProviderFilePath: false,
+		verifyDeployment: false,
+		ignoreComments: false,
+		ignoreColumnCollation: false,
+		ignoreAuthorizer: false,
+		ignoreAnsiNulls: false,
+		generateSmartDefaults: false,
+		dropStatisticsNotInSource: false,
+		dropRoleMembersNotInSource: false,
+		dropPermissionsNotInSource: false,
+		dropObjectsNotInSource: false,
+		ignoreColumnOrder: false,
+		doNotDropObjectTypes: [],
+		excludeObjectTypes: []
+	}
+};
+
+export class MockDacFxMssqlService implements vscodeMssql.IDacFxService {
+	public exportBacpac(_: string, __: string, ___: string, ____: vscodeMssql.TaskExecutionMode): Thenable<vscodeMssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public importBacpac(_: string, __: string, ___: string, ____: vscodeMssql.TaskExecutionMode): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public extractDacpac(_: string, __: string, ___: string, ____: string, _____: string, ______: vscodeMssql.TaskExecutionMode): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public createProjectFromDatabase(_: string, __: string, ___: string, ____: string, _____: string, ______: vscodeMssql.ExtractTarget, _______: vscodeMssql.TaskExecutionMode): Thenable<vscodeMssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public deployDacpac(_: string, __: string, ___: boolean, ____: string, _____: vscodeMssql.TaskExecutionMode, ______?: Record<string, string>): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public generateDeployScript(_: string, __: string, ___: string, ____: vscodeMssql.TaskExecutionMode, ______?: Record<string, string>): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public generateDeployPlan(_: string, __: string, ___: string, ____: vscodeMssql.TaskExecutionMode): Thenable<vscodeMssql.GenerateDeployPlanResult> { return Promise.resolve(mockDacFxResult); }
+	public getOptionsFromProfile(_: string): Thenable<vscodeMssql.DacFxOptionsResult> { return Promise.resolve(mockDacFxMssqlOptionResult); }
+	public validateStreamingJob(_: string, __: string): Thenable<mssql.ValidateStreamingJobResult> { return Promise.resolve(mockDacFxResult); }
+}
+
+export class MockSchemaCompareService implements vscodeMssql.ISchemaCompareService {
+	schemaCompareGetDefaultOptions(): Thenable<vscodeMssql.SchemaCompareOptionsResult> {
+		throw new Error('Method not implemented.');
+	}
+}
+
+export class MockVscodeMssqlIExtension implements vscodeMssql.IExtension {
+	sqlToolsServicePath: string = '';
+	dacFx: vscodeMssql.IDacFxService;
+	schemaCompare: vscodeMssql.ISchemaCompareService;
+	azureFunctions: vscodeMssql.IAzureFunctionsService;
+
+	constructor() {
+		this.dacFx = new MockDacFxMssqlService;
```

Comment:
You shouldn't need all these mocks since we're not using dacfx or schema compare in this extension.